scripts/superdesk-search/search.js

```diff
@@ -2050,13 +2070,14 @@
                             var itemsById = angular.extend({}, listComponent.state.itemsById);
 
                             items._items.forEach(function(item) {
-                                if (!itemsById[item._id] || !isSameVersion(itemsById[item._id], item)) {
-                                    itemsById[item._id] = item;
+                                var itemId = search.generateTrackByIdentifier(item);
```

Comment:
That's correct. Now when we lock we use the `id:version` and when an unlock happens we use `updateAllItems` function to update all previous versions to unlock. This is required in published collection where different versions of the same story is visible and when an update happens, returned version is different than the locked one.